File: frontend/src/api/subject.ts
import { apiClient } from './client';

export interface SubjectRead {
  id: string;
  name: string;
  description: string | null;
  orgId: string;
  teacherId: string;
  teacherInfo?: {
    fullName: string;
    email: string;
    phone?: string | null;
    bio?: string | null;
  };
  createdAt: string;
}

/** 
 * GET /subjects/ — получить список дисциплин организации.
 * Теперь используется официальный эндпоинт бэкенда.
 */
export async function getSubjects(): Promise<SubjectRead[]> {
  const { data } = await apiClient.get<SubjectRead[]>('/subjects/');
  return data;
}

/** 
 * GET /subjects/{id} — детали дисциплины (опционально)
 */
export async function getSubject(id: string): Promise<SubjectRead> {
  const { data } = await apiClient.get<SubjectRead>(`/subjects/${id}`);
  return data;
}

File: backend/app/schemas/subject.py
import uuid
from datetime import datetime
from typing import Optional
from app.schemas.base import APIModel

class SubjectBase(APIModel):
    name: str
    description: Optional[str] = None

class SubjectCreate(SubjectBase):
    org_id: uuid.UUID
    teacher_id: uuid.UUID  # Ответственный за предмет

class SubjectTeacherInfo(APIModel):
    full_name: str
    email: str
    phone: Optional[str] = None
    bio: Optional[str] = None

class SubjectRead(SubjectBase):
    id: uuid.UUID
    org_id: uuid.UUID
    teacher_id: uuid.UUID
    teacher_info: Optional[SubjectTeacherInfo] = None # Добавлено
    created_at: datetime
    updated_at: datetime